### PR TITLE
 refactor(app): wire deck cal status API to UI 

### PR DESCRIPTION
**Pull request by mcous**
## overview

This PR wires `GET /calibration/status` to the robot controls card and associated UI.

## changelog

-  refactor(app): wire deck cal status API to UI 

## review requests

- [ ] Deck Calibration control renders proper warning depending on good, bad, identity, or singularity status
- [ ] Robot Calibration Check control is disabled if status is bad, identity, or singularity
- [ ] Robot Calibration Check control **does not render** if status is unknown
    - Note: this also ensure it will not show up on robots without the GET /calibration/status endpoint, which is good
- [ ] `GET /calibration/status` is polled when robot page is open

Note: Tooltips are not working because out Button components _still_ can't display a tooltip if they're disabled, because disabling them sets `pointer-events: none` in CSS. I will try to follow up and I have written this UI + tests to show tooltips as soon as that is fixed. Give the code a look though for that tooltip copy.

## risk assessment

Low, but but do try to smoke test fully to validate that the unit tests are good

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
Tested, LGTM

File: app/src/calibration/api-types.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  DECK_CAL_STATUS_SINGULARITY,
+} from './constants'
+
+export type DeckCalibrationStatus =
+  | DECK_CAL_STATUS_OK
+  | DECK_CAL_STATUS_IDENTITY
+  | DECK_CAL_STATUS_BAD_CALIBRATION
+  | DECK_CAL_STATUS_SINGULARITY
+
+export type CalibrationStatus = {|
```
Comment (b-cooper) [inline app/src/calibration/api-types.js:16]: I find it a little confusing that a type called `DeckCalibrationStatus` represents a string enum, while a type called `CalibrationStatus` represents a whole block of data. I feel like `CalibrationData` or `CalibrationInfo` would be less confusing. 

I realize these two terms were technically defined in the backend implementation, I still wonder if there is a way to minimize confusion.

**Review by b-cooper — APPROVED** [review]
⛸️ Code looks good, Testing on robot now.  
Just a naming nit.

File: app/src/calibration/api-types.js (hunk 1 above)
Comment (mcous) [reply app/src/calibration/api-types.js:16]: Maybe `RobotCalibrationStatus`?

Comment (b-cooper) [reply app/src/calibration/api-types.js:16]: That sounds like a good compromise. 👍

**Pull request merged**